pixels-index/pixels-index-rocksdb/src/main/java/io/pixelsdb/pixels/index/rocksdb/RocksDBIndex.java

```diff
@@ -444,6 +488,24 @@ private boolean startsWith(byte[] array, byte[] prefix)
         return true;
     }
 
+    private Slice setLowerBound(byte[] prefixBytes)
+    {
+        ByteBuffer lowerBoundBuffer = ByteBuffer.allocate(prefixBytes.length + Long.BYTES);
+        lowerBoundBuffer.put(prefixBytes);
+        lowerBoundBuffer.putLong(0L);
+        byte[] lowerBoundBytes = lowerBoundBuffer.array();
+        return new Slice(lowerBoundBytes);
+    }
+
+    private Slice setUpperBound(byte[] prefixBytes, long timestamp)
+    {
+        ByteBuffer upperBoundBuffer = ByteBuffer.allocate(prefixBytes.length + Long.BYTES);
```

Comment:
We can directly do timestamp+1 on the keyBytes, avoiding allocating and creating new byte buffers. This is a bit dangerous but worthy. Add a notice in the Java doc saying this method overwrites the content of the input key.